### DISCUSSION TITLE
Advice for Implementing New Language Support

### DISCUSSION THREAD

**Discussion by xenia-foxtrot** [capnproto/capnproto#1989] [General]
Hello!

Because the current C# support for Cap'n Proto doesn't seem to be maintained anymore, I've decided to implement my own support for C#.

The docs for doing this [suggest to contact the mailing list for advice.](https://capnproto.org/otherlang.html#:~:text=you%20should%20e-mail%20the%20list%20before%20you%20start%20hacking.%20we%20don%E2%80%99t%20bite%2C%20and%20we%E2%80%99ll%20probably%20have%20useful%20tips%20that%20will%20save%20you%20time.%20%3A), which has apparently been migrated to here. I took a look around but I didn't see this question asked already, so I figured I might as well!

Any advice on making a new language implementation for a little upstart like me?

**Comment by kentonv** [comment]
Hi! Glad to hear you'll be working on this.

I would recommend studying the C++ and Java implementations for inspiration. The C++ implementation is, of course, the most complete implementation, while Java is a somewhat more similar language to C# so might have some worthwhile tricks.

**Reply by xenia-foxtrot** [reply]
I can certainly dig in but I do find C++ code hard to read in general. I'll see if Java has any tricks.

**Comment by ocdtrekkie** [comment]
This is awesome to hear!
